### PR TITLE
Windoze/211 maven submission

### PR DESCRIPTION
**Pull request by windoze**
#211 Fix, enable submitting Feathr JAR from Maven.

### CODE REVIEW

File: feathr_project/feathr/spark_provider/_synapse_submission.py [hunk 1]
```diff
@@ -92,21 +97,45 @@ def submit_feathr_job(self, job_name: str, main_jar_path: str = None,  main_clas
...
+                current_dir = pathlib.Path(__file__).parent.resolve()
+                main_jar_path = os.path.join(current_dir, "noop-1.0.jar")
+            else:
+                # This is a PySpark job
+                pass
+        main_jar_cloud_path = None
+        if main_jar_path:
+            # Now we have a main jar, either feathr or noop
+            if main_jar_path.startswith('abfs'):
+                main_jar_cloud_path = main_jar_path
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/spark_provider/_synapse_submission.py:130]: nit: if main_jar_path and main_jar_path.startswith('abfs'):  ?

**Comment by xiaoyongzhu** [comment]
Overall look good to me. Seems like the test failure is irrelevant.

**Comment by xiaoyongzhu** [comment]
Can we also add a bit test case to cover this?

**Review by xiaoyongzhu — APPROVED** [review]

**Comment by xiaoyongzhu** [comment]
Pasting some background info for this PR: 
```
Since Feathr is on Maven now (https://search.maven.org/artifact/com.linkedin.feathr/feathr_2.12), we should consider using Maven as the source to submit Spark jobs,rather than using the public wasb path, since it's a bit hard to maintain and doesn't distribute well (like don't have mirrors etc., and will be slow).
```

**Comment by jainr** [comment]
+1 to adding tests to cover this. Looks good to me.

**Comment by windoze** [comment]
E2E test was added in the latest commit.

**Pull request merged**